nixos/modules/security/pam.nix

```diff
@@ -262,12 +273,14 @@ let
           # prompts the user for password so we run it once with 'required' at an
           # earlier point and it will run again with 'sufficient' further down.
           # We use try_first_pass the second time to avoid prompting password twice
-          (optionalString (cfg.unixAuth && (config.security.pam.enableEcryptfs || cfg.pamMount)) ''
+          (optionalString (cfg.unixAuth && (config.security.pam.enableEcryptfs || cfg.pamMount || cfg.enableKwallet)) ''
               auth required pam_unix.so ${optionalString cfg.allowNullPassword "nullok"} likeauth
               ${optionalString config.security.pam.enableEcryptfs
                 "auth optional ${pkgs.ecryptfs}/lib/security/pam_ecryptfs.so unwrap"}
               ${optionalString cfg.pamMount
                 "auth optional ${pkgs.pam_mount}/lib/security/pam_mount.so"}
+              ${optionalString cfg.enableKwallet
+                "auth optional ${pkgs.kde5.kwallet-pam}/lib/security/pam_kwallet5.so auto_start"}
```

Comment:
Even better, it turns out that `auto_start` does absolutely nothing with pam_kwallet.  It appears that it was accidentally copied over from the pam_gnome_keyring config at some point, and that tidbit has been repeated in many places around the net ever since.